code/ui/manager/package.json

```diff
@@ -47,8 +47,8 @@
     "!src/**/*"
   ],
   "scripts": {
-    "check": "node --loader esbuild-register/loader -r esbuild-register ../../../scripts/prepare/check.ts",
-    "prep": "node --loader esbuild-register/loader -r esbuild-register ../../../scripts/prepare/esm-bundle.ts"
+    "check": "node --loader ../../../scripts/node_modules/esbuild-register/loader.js -r ../../../scripts/node_modules/esbuild-register/register.js ../../../scripts/prepare/check.ts",
+    "prep": "node --loader ../../../scripts/node_modules/esbuild-register/loader.js -r ../../../scripts/node_modules/esbuild-register/register.js ../../../scripts/prepare/esm-bundle.ts"
```

Comment:
This is pretty dang ugly.  Are you sure this isn't a good enough reason to explore using `tsx` instead?  ;)

Reply:
`tsx` will be installed in `scripts`. It would then look like this:

```json
"check": ../../../scripts/node_modules/tsx/index.js ../../../scripts/prepare/check.ts
```

This is still ugly. A bit less ugly than the current solution, but still ugly. I don't see strong reasons for now to invest even more time into this PR. 

Why should we introduce a tool, which builds on top of esbuild, if we can use esbuild directly and have no issues with it? I haven't seen strong reasons and arguments for introducing it so far.

Reply:
oh yeah that's not much better.  Seems like something that can be cleaned up later if we find a nice way.  Using `NODE_PATH` could also be an option, though I feel like it's discouraged / deprecated these days.

Reply:
Hey @IanVS 

I have [spiked](https://github.com/storybookjs/storybook/pull/24846) a bit a solution using `tsx`. I haven't invested much time, but the following occurred:

The strange thing is, that named exports are not allowed to be imported with named imports.

e.g. `fs-extra`
Currently, we do named imports like this:

```tsx
import { readJson } from 'fs-extra'; 
```

Node then complains and says:

```
import { readFile } from 'fs-extra';
         ^

SyntaxError: The requested module 'fs-extra' does not provide an export named 'readFile'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)
```

Indeed, if I go to the Readme of fs-extra the following is written:

> ESM
> There is also an fs-extra/esm import, that supports both default and named exports. However, note that fs methods are not included in fs-extra/esm; you still need to import fs and/or fs/promises separately:
> 
> ```ts
> import { readFileSync } from 'fs'
> import { readFile } from 'fs/promises'
> import { outputFile, outputFileSync } from 'fs-extra/esm'
> ```
> 
> Default exports are supported:
> 
> ```ts
> import fs from 'fs'
> import fse from 'fs-extra/esm'
> // fse.readFileSync is not a function; must use fs.readFileSync
> ```
> 

So, I have to change all named imports to default imports.

The same error then occurs on relative file imports, like this one:

```ts
import { allTemplates } from '../code/lib/cli/src/sandbox-templates';
```

```
/Users/valentinpalkovic/Projects/storybook-next/scripts/task.ts:33
  allTemplates as TEMPLATES,
  ^

SyntaxError: The requested module '../code/lib/cli/src/sandbox-templates' does not provide an export named 'allTemplates'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)
```

I am not sure what’s going on here or why the error occurs with tsx, but not with the esbuild-loader